### DISCUSSION TITLE
Estouro de Memória

### DISCUSSION THREAD

**Discussion by souzadeveloper** [balta-io/forum#362] [Tira Dúvidas] [answered]
Bom dia Balta, tudo bem? Existe alguma regra para se trabalhar com arquivos em Serviços do Windows usando C#. Percebo que todas as vezes que carrego arquivos em um MemoryStream usando Serviços, meu serviço começa a usar muita memória e isso só vai aumentando conforme o tempo em que o Serviço fica rodando chegando a casos de dar o erro System.OutOfMemoryException.

Já tentei usar using e até Idisposible nas classes e não melhora nada.... E agora estou com uma situação onde vou precisar trabalhar com muitos arquivos mesmo... Só para resumir, o Serviço lê vários arquivos, compacta e envia para o S3 da Amazon. A ideia era que no final de cada processamento a memória fosse liberada, mas não só vai acumulando a cada processamento.

Agradeço qualquer ajuda.

**Comment by andrebaltieri** [comment, marked as answer]
Opa, existe sim! Conforme comento no curso, se você tentar abrir um arquivo muito grande de uma vez só, vai dar problema.
O ideal ao se trabalhar com arquivos grandes é ler em pedaços (chunks) ou linha a linha.

O BufferedStream pode ajudar nisso:
https://learn.microsoft.com/en-us/dotnet/api/system.io.bufferedstream?view=net-7.0&redirectedfrom=MSDN

**Comment by souzadeveloper** [comment]
Valeu Balta. Obrigado!
